src/shelly_switch.cpp

```diff
@@ -138,6 +139,10 @@ Status ShellySwitch::SetConfig(const std::string &config_json,
   cfg_->initial_state = cfg.initial_state;
   cfg_->auto_off = cfg.auto_off;
   cfg_->auto_off_delay = cfg.auto_off_delay;
+  if (cfg_->state_led_en != cfg.state_led_en) {
```

Comment:
i think we should disallow setting to any value if it's currently -1.

Reply:
Like this?
```suggestion
  if (cfg_->state_led_en != -1 && cfg_->state_led_en != cfg.state_led_en) {
```

Reply:
even earlier, during validation, we should check for valid new values - 0, 1 and that previous not -1, and send an error

Reply:
would you like add append an commit with a change according to your ideas

Reply:
i don't think i can push directly to this branch, but here's what i meant:

```
  if ((cfg_->state_led_en == -1 && cfg.state_led_en != -1) ||
      (cfg_->state_led_en != -1 && cfg.state_led_en != 0 &&
       cfg.state_led_en != 1)) {
    return mgos::Errorf(STATUS_INVALID_ARGUMENT, "invalid %s", "state_led_en");
  }
```

Reply:
updated my branch